ophyd/controls/positioner.py

```diff
@@ -425,8 +421,6 @@ def __init__(self, setpoint, readback=None,
                                         name=self.name))
 
             self._readback.subscribe(self._pos_changed)
-
-            self._set_position(self._readback.value)
```

Comment:
If unconnected, I think it should try an explicit `readback.get()` to wait on the connection. That way it has a chance of getting the position as requested -- or it'll throw a `TimeoutError` with your new changes.